win32_event_log/tests/test_check.py

```diff
@@ -28,6 +31,40 @@ def test_expected(aggregator, dd_run_check, new_check, instance, report_event, s
     )
 
 
+def test_recover_from_broken_subscribe(aggregator, dd_run_check, new_check, instance, event_reporter, caplog):
+    """
+    Test the check can recover from a broken EvtSubscribe handle
+
+    Issue originally surfaced when the event publisher is unregistered while we
+    have an EvtSubscribe handle to one of it's channels. This is difficult to test
+    here so we mimic it by replacing the subscription handle.
+    """
+    # Speed up test
+    instance['timeout'] = 0.1
+    check = new_check(instance)
+
+    # Run check_initializations to create EvtSubscribe
+    dd_run_check(check)
+
+    # Create an event
+    event_reporter.report('message').join()
+
+    # Mutate the subscription handle so that the check's EvtNext() fails
+    check._subscription = None
```

Comment:
You can also try to EvtClose on it, I think, to be bit more realistic (I wonder)